frontend/src/container/PublicProfile.jsx

```diff
@@ -371,12 +371,14 @@ export class PublicProfile extends React.Component {
     }
   }
 
-  isPublicProfileEditable = (connectedUser, publicProfileId, profileObject) => {
+  isPublicProfileEditable = (connectedUser, publicProfileId, profileObject, schemaObject) => {
```

Comment:
seeing the usage of the function in CoverImage and Avatar makes me realize that the function now to more that what it's supposed to do.

As seen with you, I advise to split the function into to parts, one to check the users ids and profile and one to check if the schema is empty.
And use const in the render to put the different values for the 3 usages:
- CoverImage and Avatar which are the same value
- Information
- Personal Page

Reply:
updated